proxy/proxy.go

```diff
@@ -12,6 +13,10 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const (
+	TenantPrefixKey = "tenant-prefix"
```

Comment:
normally we try to keep the context keys unexported to prevent this from being abused. If we write a `WithPrefix` context setter func in this package then we can make this unexported

Reply:
Cool, I was thinking maybe that would be an overkill but combined with the above comment it makes a lot of sense.